Database documentation/MongoDB.md

```diff
@@ -0,0 +1,46 @@
+# Database
+
+## MongoDB
+
+MongoDB is een open source document-georiënteerde database en is geschreven in C++. Er is geen schema, de documenten worden in de vorm van BSON (binair JSON) opgeslagen en de structuur van deze documenten is flexibel. De database kan gemakkelijk gedistribueerd worden, de data wordt dan over meerdere computers verspreid om gedistribueerde gegevensverwerking mogelijk te maken. MongoDB is geen relationeel databasemanagementsysteem, er is geen ondersteuning voor joins en voldoet ook niet aan de ACID-regels want de ondersteuning voor transacties is beperkt. MongoDB wordt gerekend tot de zogenaamde NoSQL-databases.
+
+MongoDb is geschikt voor grote data het is schaalbaar en werk zonder relaties in tegenstelling tot bijvoorbeeld SQLite. Omdat we in ons project geen echte relaties nodig hebben is de keuze daarom ook makkelijk gemaakt voor MongoDB.
+
+### Voorbeeld eventuele database structuur
+
+```json
+{
+ "user_id": "1",
+ "username": "Maliek Meersschaert",
+ "password": "SOME HASH",
+ "polls": [
+     {
+      "poll_id": "1",
+      "name": "Amerikaanse verkiezingen",
+      "options": ["Donald Trump", "Hilary Clinton", "Gary Johnson", "Jill Stein"],
+      "values": [10,9,2,1]
+    },
+    {
+      "poll_id": "2",
+      "name": "Lievelingskleur",
+      "options": ["Geel", "Blauw"],
+      "values": [1,2]
+    }
+  ]
+}
+ ```
```

Comment:
Ik ben persoonlijk meer voor de eerste structuur. ik zou wel ook een user_id toevoegen aan de poll objecten (als er een user is). Dat gaat later gemakkelijker zijn om de bijpassende user te vinden voor een poll. anders gaan we denk ik alle users moeten overlopen en kijken of ze die poll id bevatten.